warehouse/query-core/src/main/java/datawave/query/config/KeywordQueryConfiguration.java

```diff
@@ -0,0 +1,162 @@
+    @SuppressWarnings("unused")
+    private static final long serialVersionUID = 1678850178943683419L;
+
+    private transient Collection<Range> ranges;
+
+    int minNgrams = YakeKeywordExtractor.DEFAULT_MIN_NGRAMS;
+    int maxNgrams = YakeKeywordExtractor.DEFAULT_MAX_NGRAMS;
+    int maxKeywords = YakeKeywordExtractor.DEFAULT_KEYWORD_COUNT;
+    float maxScore = YakeKeywordExtractor.DEFAULT_MAX_SCORE_THRESHOLD;
+    int maxContentChars = YakeKeywordExtractor.DEFAULT_MAX_CONTENT_LENGTH;
+
+    public KeywordQueryConfiguration() {
+        super();
+        setQuery(new QueryImpl());
+    }
+
+    public KeywordQueryConfiguration(BaseQueryLogic<?> configuredLogic, Query query) {
+        super(configuredLogic);
+        setQuery(query);
+        this.ranges = new TreeSet<>();
+    }
+
+    public KeywordQueryConfiguration(KeywordQueryConfiguration other) {
+        copyFrom(other);
+    }
+
+    public void copyFrom(KeywordQueryConfiguration other) {
+        super.copyFrom(other);
+
+        this.setMinNgrams(other.minNgrams);
+        this.setMaxNgrams(other.maxNgrams);
+        this.setMaxKeywords(other.maxKeywords);
+        this.setMaxScore(other.maxScore);
+        this.setMaxContentChars(other.maxContentChars);
+    }
+
+    public int getMaxContentChars() {
+        return maxContentChars;
+    }
+
+    public void setMaxContentChars(int maxContentChars) {
+        this.maxContentChars = maxContentChars;
+    }
+
+    public int getMaxKeywords() {
+        return maxKeywords;
+    }
+
+    public void setMaxKeywords(int maxKeywords) {
+        this.maxKeywords = maxKeywords;
+    }
+
+    public int getMaxNgrams() {
+        return maxNgrams;
+    }
+
+    public void setMaxNgrams(int maxNgrams) {
+        this.maxNgrams = maxNgrams;
+    }
+
+    public float getMaxScore() {
+        return maxScore;
+    }
+
+    public void setMaxScore(float maxScore) {
+        this.maxScore = maxScore;
+    }
+
+    public int getMinNgrams() {
+        return minNgrams;
+    }
+
+    public void setMinNgrams(int minNgrams) {
+        this.minNgrams = minNgrams;
+    }
+
+    /**
+     * Factory method that instantiates a fresh KeywordQueryConfiguration
+     *
+     * @return - a clean KeywordQueryConfiguration
+     */
+    public static KeywordQueryConfiguration create() {
+        return new KeywordQueryConfiguration();
+    }
+
+    public synchronized void addRange(final Range range) {
+        if (null != range) {
+            this.ranges.add(range);
+        }
+    }
+
+    public synchronized Collection<Range> getRanges() {
+        return new ArrayList<>(this.ranges);
+    }
+
+    public synchronized void setRanges(final Collection<Range> ranges) {
+        // As a single atomic operation, clear the range and add all of the
+        // specified ranges
+        this.ranges.clear();
+        if (null != ranges) {
+            this.ranges.addAll(ranges);
+        }
+    }
```

Comment:
I don't see any threaded access to this, probably doesn't need to be synchronized?

Reply:
Moved these into the KeywordQueryState class.